### PR TITLE
RCD context actions menu

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[QOL]
## About the PR 
Changes the RCD to use a context actions menu
image
I haven't included the pod door modes as they seem to be entirely unused.

## Why's this needed? 
It was crying out for context actions.

## Changelog

```changelog
(u)LeahTheTech
(+)The RCD now uses a context actions menu, allowing you to immediately switch to deconstruct slightly faster!
```

### CODE REVIEW

File: goonstation.dme [hunk 1]
```diff
@@ -30,6 +30,7 @@ var/datum/preMapLoad/preMapLoad = new
 #include "code\__secret_public.dme"
 
 // BEGIN_INCLUDE
+#include "_std\defines\RCD.dm"
```
Comment (frawhst) [inline goonstation.dme:33]: This should be in `__std.dme`, not here

File: code/obj/item/rcd.dm [hunk 2]
```diff
@@ -190,10 +185,15 @@ Broken RCD + Effects
 				boutput(user, "This cartridge is not made of the proper material to be used in \The [src].")
 
 	attack_self(mob/user as mob)
+		user.showContextActions(src.contextActions, src, src.contextLayout)
+
+	proc/switch_mode(var/mode, var/mob/user)
+		if (!(mode in src.modes))
+			return
```
Comment (frawhst) [inline code/obj/item/rcd.dm:192]: This should probably `CRASH` with some debug info- we want to know if something is passing nonsense as an RCD mode (or maybe someone just made a typo, etc)

File: code/obj/item/rcd.dm [hunk 3]
```diff
@@ -161,6 +151,11 @@ Broken RCD + Effects
 
 	New()
 		..()
+		src.contextActions = list()
+		for(var/actionType in childrentypesof(/datum/contextAction/rcd)) //see context_actions.dm for those
+			var/datum/contextAction/rcd/action = new actionType()
+			if (action.mode in src.modes)
+				src.contextActions += action
 		src.UpdateIcon()
 		return
```
Comment (frawhst) [inline code/obj/item/rcd.dm:160]:
```suggestion
```
unnecessary

File: code/obj/item/rcd.dm [hunk 4]
```diff
@@ -219,9 +219,6 @@ Broken RCD + Effects
 			if (RCD_MODE_LIGHTTUBES)
 				boutput(user, "Changed mode to 'Light Tube Fixture'")
 
-		// Gonna change this so it doesn't shit sparks when mode switched
-		// Just that it does it only after actually doing something
-		//src.shitSparks()
 		src.UpdateIcon()
 		return
```
Comment (frawhst) [inline code/obj/item/rcd.dm:223]:
```suggestion
```
god why are there so many of these everywhere

**Review by frawhst — COMMENTED** [review]
Looks good, much easier to use

File: code/obj/item/rcd.dm (hunk 4 above)
Comment (TobleroneSwordfish) [reply code/obj/item/rcd.dm:223]: very important, must be certain the proc returns, defensive programming

**Pull request merged**